src/core/elements/stomp_boots.rs

```diff
@@ -167,13 +182,29 @@ fn update(
 fn update_wearer(
     entities: Res<Entities>,
     mut commands: Commands,
-    wearing_stomp_boots: Comp<WearingStompBoots>,
+    wearing_stomp_boots: CompMut<WearingStompBoots>,
+    mut worn_stomp_boots: CompMut<WornStompBoots>,
     player_indexes: Comp<PlayerIdx>,
     collision_world: CollisionWorld,
     kinematic_bodies: Comp<KinematicBody>,
+    killed_players: Comp<PlayerKilled>,
+    mut hydrated: CompMut<MapElementHydrated>,
+    spawners: Comp<DehydrateOutOfBounds>,
     transforms: Comp<Transform>,
 ) {
-    for (entity, _) in entities.iter_with(&wearing_stomp_boots) {
+    for (entity, WearingStompBoots { stomp_boots }) in entities.iter_with(&wearing_stomp_boots) {
+        if killed_players.get(entity).is_some() {
+            // Respawn the boots worn by the player who was just killed
+            if let Some(spawner) = spawners.get(*stomp_boots) {
+                if worn_stomp_boots.get(*stomp_boots).is_some() {
+                    worn_stomp_boots.remove(*stomp_boots);
+                    hydrated.remove(**spawner);
+                }
+            }
+            // We don't need to proceed because the player is dead
+            return;
```

Comment:
I think this should be `continue` instead of `return`. This would prevent update for a second player who is wearing boots - but we only want to skip this iteration for this killed player.

Otherwise this change looks great, tested it out seems to be working. Thank you!

Reply:
> I think this should be `continue` instead of `return`. This would prevent update for a second player who is wearing boots - but we only want to skip this iteration for this killed player.

I forgot it's not bevy :/

However, after experimenting for a while, I discovered that players can equip two boot items, but only one of them will respawn when the player is killed. I suggest dropping the worn boots when a player attempts to equip another pair.